protocol/x/vault/keeper/orders_test.go

```diff
@@ -299,35 +299,48 @@ func TestRefreshVaultClobOrders(t *testing.T) {
 			if tc.expectedErr != nil {
 				// Verify that no order is placed and chain doesn't halt.
 				require.Empty(t, tApp.App.ClobKeeper.GetAllStatefulOrders(ctx))
-				tApp.AdvanceToBlock(uint32(tApp.GetBlockHeight())+1, testapp.AdvanceToBlockOptions{})
+				tApp.AdvanceToBlock(uint32(tApp.GetBlockHeight())+12, testapp.AdvanceToBlockOptions{})
 				return
 			}
 
-			// Helper function that verifies that most recent client IDs up-to-date with vault orders.
-			verifyMostRecentClientIds := func() {
+			// Helper function that verifies that most recent client IDs are up-to-date with vault orders.
+			verifyMostRecentClientIds := func(expectedClientIds []uint32) {
+				// Verify that most recent client IDs are as expected.
 				mostRecentClientIds := tApp.App.VaultKeeper.GetMostRecentClientIds(ctx, tc.vaultId)
+				require.Equal(t, expectedClientIds, mostRecentClientIds)
+
+				// Verify that stateful order IDs have expected client IDs.
 				allStatefulOrders := tApp.App.ClobKeeper.GetAllStatefulOrders(ctx)
 				for i, order := range allStatefulOrders {
-					require.Equal(t, mostRecentClientIds[i], order.OrderId.ClientId)
+					require.Equal(t, expectedClientIds[i], order.OrderId.ClientId)
 				}
 			}
+			// Get canonical and flipped client IDs of this vault's orders.
```

Comment:
nit: Break up into more tests to explicitly test either the replace after expire and the replace after price / size change rather than having a single large test.